src/permission-matcher.js

```diff
@@ -95,10 +95,11 @@ class PermissionMatcher {
         let regex = '^' + str_replace('[^/]+', '\\*', preg_quote(pattern, '~')) + '$';
         regex = str_replace('\\*', '(.*)', regex);
         regex = new RegExp(regex, 'i');
-        
-        const invert = pattern.startsWith('!');
+
+        const token = '!';
+        const invert = pattern.substr(0, token.length) === token;
```

Comment:
`startsWith` uses under the hood `indexOf` so it's common practice to use that instead.